contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,486 @@
+     *  @param _claimId The Id of the active claim in Vault contract.
+     *  @param _evidence URI of the evidence to support the challenge.
+     *  Note that the validity of the claim should be checked by Hat arbitrator.
+     *  @return surplusFee The amount of fee that exceeds required arbitration cost.
+     */
+    function notifyArbitrator(bytes32 _claimId, string calldata _evidence) external payable returns (uint256 surplusFee) {
+        require(msg.sender == address(hatArbitrator), "Wrong caller");
+        require(!claimChallenged[_claimId], "Claim already challenged");
+
+        uint256 arbitrationCost = getArbitrationCost();
+        require(msg.value >= arbitrationCost, "Should pay the full deposit.");
+        surplusFee = msg.value - arbitrationCost;
+
+        claimChallenged[_claimId] = true;
+
+        uint256 localDisputeId = disputes.length;
+
+        DisputeStruct storage dispute = disputes.push();
+        dispute.claimId = _claimId;
+
+        // Preemptively create a new funding round for future appeals.
+        dispute.rounds.push();
+
+        uint256 externalDisputeId = klerosArbitrator.createDispute{value: arbitrationCost}(RULING_OPTIONS,  arbitratorExtraData);
+        dispute.externalDisputeId = externalDisputeId;
+        externalIDtoLocalID[externalDisputeId] = localDisputeId;
+
+        emit Challenged(_claimId);
+        emit Dispute(klerosArbitrator, externalDisputeId, metaEvidenceUpdates, localDisputeId);
+        emit Evidence(klerosArbitrator, localDisputeId, msg.sender, _evidence);
+    } 
+
+    /** @dev Give a ruling for a dispute. Can only be called by the arbitrator.
+     *  @param _disputeId ID of the dispute in the arbitrator contract.
+     *  @param _ruling Ruling given by the arbitrator. Note that 0 is reserved for "Refused to arbitrate".
+     */
+    function rule(uint256 _disputeId, uint256 _ruling) external override {
+        uint256 localDisputeId = externalIDtoLocalID[_disputeId];
+
+        DisputeStruct storage dispute = disputes[localDisputeId];
+
+        require(!dispute.resolved, "Already resolved");
+        require(_ruling <= RULING_OPTIONS, "Invalid ruling option");
+        require(address(klerosArbitrator) == msg.sender, "Only the arbitrator can execute");
+        uint256 finalRuling = _ruling;
+    
+        // If one side paid its fees, the ruling is in its favor. Note that if the other side had also paid, an appeal would have been created.
+        Round storage round = dispute.rounds[dispute.rounds.length - 1];
+        if (round.fundedSides.length == 1) finalRuling = round.fundedSides[0];
+
+        dispute.ruling = Decision(finalRuling);
+        dispute.resolved = true;
+
+        bytes32 claimId = dispute.claimId;
+        if (finalRuling == uint256(Decision.DismissResolution)) {            
+            hatArbitrator.dismissResolution(vault, claimId); //
+        } else {
+            // Arbitrator sided with committee or refused to arbitrate (gave 0 ruling).
```

Comment:
I'm not sure if we want to dismiss or execute the resolution if arbitrator doesn't give a decision. @jellegerbrandy ?